### PR TITLE
use asciidoclet to improve javadocs readability

### PR DESCRIPTION
**Pull request by romain-grecourt**
1) Add support for asciidoclet:
  The version of asciidoclet that supports java9+ is 1.5.6. Unfortunately that version
   is not available in Maven central but only in bintray for unkown reason. I filed a
    bug on Github, see https://github.com/asciidoctor/asciidoclet/issues/93.
  Added configuration for the bintray repository inside the javadoc profile in order not
   to impact the rest of the project.
  Added configuration for the doclet in the default profile. This means that one must
   use the javadoc profile in order to build any of our javadocs, otherwise it will fail
   with some artifact resolution error. Added a workaround for jruby not in the doclet path
   which basically consists in declaring jruby as a docletArtifact to add in the classpath
   for the doclet.

2) Updated OidcConfig to use asciidoclet
   Updated the class javadoc for OidcConfig to show what using asciidoclet would
   look like.

### CODE REVIEW

**Comment by romain-grecourt** [comment]
Seems like I have to push the timeout because using the doclet slows down the build significantly. It might be a blocker and we may be forced to re-think our pipelines before introducing something like this.

**Comment by ljnelson** [comment]
> The build failure is due to some obscure Maven behavior (bug!) where the code that resolves the artifact for the doclet does not have any additional repository configured. It only happens when doing a build with multiple modules in the reactor.
> 
> Forcing the artifacts to be resolved as a project dependency or plugin dependency does not work since the plugin tries to resolve it from Maven Central.

I see you attempted to work around all this at one point with a `<repository>` entry.  Did you try (a) adding the doclet as a plugin dependency while also (b) adding a [`<pluginRepository>` entry](https://maven.apache.org/pom.html#Plugin_Repositories)?  I don't know if this will work or not; I'd actually hope it doesn't!

File: javadocs/pom.xml [hunk 1]
```diff
@@ -37,6 +37,13 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <executions>
+                    <!-- skip the execution added by the javadoc profile -->
```
Comment (ljnelson) [inline javadocs/pom.xml]: 👍

File: pom.xml [hunk 2]
```diff
@@ -215,8 +215,8 @@
                     <artifactId>maven-compiler-plugin</artifactId>
                     <version>${version.plugin.compiler}</version>
                     <configuration>
-                        <source>${java.version}</source>
-                        <target>${java.version}</target>
+                        <source>1.8</source>
```
Comment (ljnelson) [inline pom.xml]: It occurs to me that if you want to keep the `${java.version}` property, you could probably write a `maven-enforcer-plugin` rule (or [use a pre-existing one](https://maven.apache.org/enforcer/enforcer-rules/requireProperty.html)) to derive the javadoc API version from it, or vice versa.  Once we go to Java 9+ throughout, the various `1.x` flavors will go away and `9` will work everywhere.  I am not sure it's worth it now, so I support the hard coding you're doing here.

**Comment by romain-grecourt** [comment]
> > The build failure is due to some obscure Maven behavior (bug!) where the code that resolves the artifact for the doclet does not have any additional repository configured. It only happens when doing a build with multiple modules in the reactor.
> > Forcing the artifacts to be resolved as a project dependency or plugin dependency does not work since the plugin tries to resolve it from Maven Central.
> 
> I see you attempted to work around all this at one point with a `<repository>` entry. Did you try (a) adding the doclet as a plugin dependency while also (b) adding a [`<pluginRepository>` entry](https://maven.apache.org/pom.html#Plugin_Repositories)? I don't know if this will work or not; I'd actually hope it doesn't!

Yes,  I tried adding a pluginRepository and plugin dependencies. I've debugged the code that failed to resolve the artifact and it didn't have any additional repository configured :(

**Comment by ljnelson** [comment]
> Yes, I tried adding a pluginRepository and plugin dependencies. I've debugged the code that failed to resolve the artifact and it didn't have any additional repository configured :(

Huh.  Well, the good news is that the doclet appears to be in Maven Central as of November 8, 2018: https://search.maven.org/artifact/org.asciidoctor/asciidoclet/1.5.6/jar

**Comment by romain-grecourt** [comment]
Closing as agreed with the team.

**Pull request closed without merging**